src/arviz_plots/plots/pair_plot.py

```diff
@@ -0,0 +1,495 @@
+                "ylabel",
+                index=0,
+                data=distribution,
+                ignore_aes=ylabel_ignore,
+                labeller=labeller,
+                subset_info=True,
+                **ylabel_kwargs,
+            )
+    elif marginal and triangle == "upper":
+        diag_xlabel_kwargs = copy(visuals.get("diag_xlabel", {}))
+
+        if diag_xlabel_kwargs is not False:
+            _, _, diag_xlabel_ignore = filter_aes(
+                plot_matrix, aes_by_visuals, "diag_xlabel", sample_dims
+            )
+            plot_matrix.map(
+                labelled_x,
+                "diag_xlabel",
+                subset_info=True,
+                ignore_aes=diag_xlabel_ignore,
+                labeller=labeller,
+                **diag_xlabel_kwargs,
+            )
+
+    # set ticklabel visibility
+    set_ticklabel_visibility_kwargs = {}
+    if triangle == "upper":
+        _, _, set_ticklabel_visibility_ignore = filter_aes(
+            plot_matrix, aes_by_visuals, "set_ticklabel_visibility", sample_dims
+        )
+        plot_matrix.map(
+            set_ticklabel_visibility,
+            "set_ticklabel_visibility",
+            axis="x",
+            visible=True,
+            ignore_aes=set_ticklabel_visibility_ignore,
+            **set_ticklabel_visibility_kwargs,
+        )
+
+        if not marginal and remove_axis_bool is False:
+            plot_matrix.map(
+                set_ticklabel_visibility,
+                "set_ticklabel_visibility",
+                axis="y",
+                visible=True,
+                ignore_aes=set_ticklabel_visibility_ignore,
+                **set_ticklabel_visibility_kwargs,
+            )
+
+    # remove_axis
+    if remove_axis_bool:
+        remove_axis_kwargs = {}
+        remove_axis_kwargs["axis"] = "y"
+        _, _, remove_axis_ignore = filter_aes(
+            plot_matrix, aes_by_visuals, "remove_axis", sample_dims
+        )
+
+        # remove axis from lower triangle
+        if triangle in {"both", "lower"}:
+            plot_matrix.map_triangle(
+                remove_matrix_axis,
+                "remove_axis",
+                triangle="lower",
+                ignore_aes=remove_axis_ignore,
+                **remove_axis_kwargs,
+            )
+
+        # remove axis from upper triangle
+        if triangle in {"both", "upper"}:
+            plot_matrix.map_triangle(
+                remove_matrix_axis,
+                "remove_axis",
+                triangle="upper",
+                ignore_aes=remove_axis_ignore,
+                **remove_axis_kwargs,
+            )
+
+        # remove axis from diagonal
+        plot_matrix.map(
+            remove_axis,
+            "remove_axis",
+            ignore_aes=remove_axis_ignore,
+            **remove_axis_kwargs,
+        )
+
+    # default removal of axis for better visualization
+    default_remove_axis_kwargs = {}
+    default_remove_axis_kwargs["axis"] = "both"
```

Comment:
There is nothing where we are using this as dict, it is much easier to follow the code if instead you replace the `**default_remove_axis_kwargs` for `axis="both"`.